R/ancombc_prep.R

```diff
@@ -41,7 +41,15 @@ tse_construct = function(data, assay_name, tax_level, phyloseq) {
             tax_level = "ASV"
             tse_alt = tse
         } else {
-            tse_alt = mia::agglomerateByRank(tse, tax_level)
+            # Check if tax_level parameter belongs to taxonomyRanks
+            if (is.character(tax_level) && length(tax_level) == 1 && tax_level %in% mia::taxonomyRanks(tse)) {
+                #Merge using agglomerateByRank
+                tse_alt <- mia::agglomerateByRank(tse, tax_level)
+            } else {
+                # Merge using mergeRows
+                tse_alt <- mia::mergeRows(tse, tax_level)
+            }
+            tse_alt
```

Comment:
See above - unnecessary checks given that mia functions should be able to handle these cases automatically. Then this function can just call `mergeFeatures()` and let mia pkg do the rest. Unless I am mistaken (see mia issue #389 https://github.com/microbiome/mia/pull/389 and the last comment).

Also: agglomerateByRank and mergeRows are now deprecated in mia. They are now mergeFeaturesByRank and mergeFeatures, respectively.

Reply:
ACK.